pkg/util/exec/exec.go

```diff
@@ -67,6 +70,44 @@ func (*CommandExecutor) ExecuteCommand(debug bool, actionName string, command st
 	return nil
 }
 
+// ExecuteCommandWithTimeout starts a process and wait for its completion with timeout.
+func (*CommandExecutor) ExecuteCommandWithTimeout(debug bool, timeout time.Duration, actionName string, command string, arg ...string) (string, error) {
+	logCommand(debug, command, arg...)
+	cmd := exec.Command(command, arg...)
+
+	var b bytes.Buffer
+	cmd.Stdout = &b
+	cmd.Stderr = &b
+
+	if err := cmd.Start(); err != nil {
+		return "", createCommandError(err, actionName)
+	}
+
+	done := make(chan error, 1)
+	go func() {
+		done <- cmd.Wait()
+	}()
+
+	select {
+	case <-time.After(timeout):
+		logger.Infof("Timeout waiting for process %s to return. Killing process", command)
+		var e error
+		if err := cmd.Process.Kill(); err != nil {
```

Comment:
i'm still wondering if we should do a less harsh attempt to stop the timed out process first by sending `SIGINT` or `SIGTERM` first, before the immediate kill.

Reply:
ok ill do this